js/src/jsonrpc/generator/build-markdown.js

```diff
@@ -45,25 +66,253 @@ function formatType (obj) {
   return obj;
 }
 
-Object.keys(interfaces).sort().forEach((group) => {
-  let content = '';
+const rpcReqTemplate = {
+  method: 'web3_clientVersion',
+  params: [],
+  id: 1,
+  jsonrpc: '2.0'
+};
+
+// Checks if the value passed in is a DUMMY object placeholder for `{ ... }``
+function isDummy (val) {
+  return val === DUMMY;
+}
 
-  preamble = `${preamble}\n- [${group}](#${group})`;
-  markdown = `${markdown}\n## ${group}\n`;
+const { isArray } = Array;
 
-  Object.keys(interfaces[group]).sort().forEach((iname) => {
-    const method = interfaces[group][iname];
-    const name = `${group}_${iname}`;
-    const deprecated = method.deprecated ? ' (Deprecated and not supported, to be removed in a future version)' : '';
-    const desc = `${method.desc}${deprecated}`;
-    const params = method.params.map(formatType).join('\n');
-    const returns = formatType(method.returns);
+// Checks if the value passed is a plain old JS object
+function isObject (val) {
+  return val != null && val.constructor === Object;
+}
 
-    markdown = `${markdown}\n- [${name}](#${name})`;
-    content = `${content}### ${name}\n\n${desc}\n\n#### parameters\n\n${params || 'none'}\n\n#### returns\n\n${returns || 'none'}\n\n`;
+// Checks if a field definition has an example,
+// or describes an object with fields that recursively have examples of their own,
+// or is optional.
+function hasExample ({ optional, example, details } = {}) {
+  if (optional || example !== undefined) {
+    return true;
+  }
+
+  if (details !== undefined) {
+    const values = Object.keys(details).map((key) => details[key]);
+
+    return values.every(hasExample);
+  }
+
+  return false;
+}
+
+// Remove all optional (trailing) params without examples from an array
+function removeOptionalWithoutExamples (arr) {
+  return arr.filter(({ optional, example, details }) => {
+    return !optional || example !== undefined || details !== undefined;
   });
+}
+
+// Grabs JSON compatible
+function getExample (obj) {
+  if (Array.isArray(obj)) {
```

Comment:
Any reason to do `const { isArray } = Array` above and then directly using `Array.isArray` here? (Just feels slightly inconsistent in use)